### PR TITLE
orchestration mode code optimization .

### PR DESCRIPTION
**Pull request by yu199195**
this is orchestration mode code optimization .

### CODE REVIEW

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-configuration/src/main/java/org/apache/shardingsphere/orchestration/center/yaml/swapper/OrchestrationConfigurationYamlSwapper.java [hunk 1]
```diff
@@ -40,7 +40,7 @@
      */
     @Override
     public YamlOrchestrationConfiguration swap(final OrchestrationConfiguration configuration) {
-        Map<String, YamlCenterRepositoryConfiguration> yamlInstanceConfigurationMap = new HashMap();
+        Map<String, YamlCenterRepositoryConfiguration> yamlInstanceConfigurationMap = new HashMap<>();
```
Comment (kimmking) [inline sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-configuration/src/main/java/org/apache/shardingsphere/orchestration/center/yaml/swapper/OrchestrationConfigurationYamlSwapper.java]: why?

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-zookeeper-curator/src/main/java/org/apache/shardingsphere/orchestration/center/instance/CuratorZookeeperCenterRepository.java [hunk 2]
```diff
@@ -228,7 +229,7 @@ public void watch(final String key, final DataChangedEventListener dataChangedEv
             }
             DataChangedEvent.ChangedType changedType = getChangedType(event);
             if (DataChangedEvent.ChangedType.IGNORED != changedType) {
-                dataChangedEventListener.onChange(new DataChangedEvent(data.getPath(), null == data.getData() ? null : new String(data.getData(), "UTF-8"), changedType));
+                dataChangedEventListener.onChange(new DataChangedEvent(data.getPath(), null == data.getData() ? null : new String(data.getData(), StandardCharsets.UTF_8), changedType));
```
Comment (kimmking) [inline sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-zookeeper-curator/src/main/java/org/apache/shardingsphere/orchestration/center/instance/CuratorZookeeperCenterRepository.java]: unnecessary.
a local static field is better.

File: sharding-orchestration/sharding-orchestration-core/sharding-orchestration-core-common/src/main/java/org/apache/shardingsphere/orchestration/core/common/CenterType.java [hunk 3]
```diff
@@ -41,6 +42,8 @@
      * @return OrchestrationType enum, return {@code null} if not found
      */
     public static CenterType findByValue(final String value) {
-        return Arrays.stream(CenterType.values()).filter(each -> each.getValue().equals(value)).findFirst().get();
+        return Arrays.stream(CenterType.values())
+                .filter(each -> each.getValue().equals(value)).findFirst()
+                .orElseThrow(() -> new OrchestrationException("now only support :{},{}", "registry_center", "config_center"));
```
Comment (kimmking) [inline sharding-orchestration/sharding-orchestration-core/sharding-orchestration-core-common/src/main/java/org/apache/shardingsphere/orchestration/core/common/CenterType.java]: good.
but static field referred better than literal string.

File: sharding-orchestration/sharding-orchestration-core/sharding-orchestration-core-configcenter/src/main/java/org/apache/shardingsphere/orchestration/core/configcenter/ConfigCenter.java [hunk 4]
```diff
@@ -180,13 +180,12 @@ private void persistShardingSchemaName(final String shardingSchemaName) {
             repository.persist(node.getSchemaPath(), shardingSchemaName);
             return;
         }
-        List<String> schemaNameList = Splitter.on(",").splitToList(shardingSchemaNames);
+        List<String> schemaNameList = Lists.newArrayList(Splitter.on(",").split(shardingSchemaNames));
```
Comment (kimmking) [inline sharding-orchestration/sharding-orchestration-core/sharding-orchestration-core-configcenter/src/main/java/org/apache/shardingsphere/orchestration/core/configcenter/ConfigCenter.java]: why?

File: sharding-orchestration/sharding-orchestration-core/sharding-orchestration-core-configcenter/src/main/java/org/apache/shardingsphere/orchestration/core/configcenter/ConfigCenter.java [hunk 5]
```diff
@@ -237,9 +236,9 @@ public boolean isShadowRule(final String shardingSchemaName) {
      */
     @SuppressWarnings("unchecked")
     public Map<String, DataSourceConfiguration> loadDataSourceConfigurations(final String shardingSchemaName) {
-        Map<String, YamlDataSourceConfiguration> result = (Map) YamlEngine.unmarshal(repository.get(node.getDataSourcePath(shardingSchemaName)));
+        Map<String, YamlDataSourceConfiguration> result = (Map<String, YamlDataSourceConfiguration>) YamlEngine.unmarshal(repository.get(node.getDataSourcePath(shardingSchemaName)));
```
Comment (kimmking) [inline sharding-orchestration/sharding-orchestration-core/sharding-orchestration-core-configcenter/src/main/java/org/apache/shardingsphere/orchestration/core/configcenter/ConfigCenter.java]: unnecessary.

File: sharding-orchestration/sharding-orchestration-core/sharding-orchestration-core-configcenter/src/main/java/org/apache/shardingsphere/orchestration/core/configcenter/ConfigCenter.java (hunk 4 above)
Comment (yu199195) [reply sharding-orchestration/sharding-orchestration-core/sharding-orchestration-core-configcenter/src/main/java/org/apache/shardingsphere/orchestration/core/configcenter/ConfigCenter.java]: splitToList return readOnly List,and Then the list will add,

File: sharding-orchestration/sharding-orchestration-core/sharding-orchestration-core-common/src/main/java/org/apache/shardingsphere/orchestration/core/common/CenterType.java [hunk 6]
```diff
@@ -31,7 +32,7 @@
     
     REGISTRY_CENTER("registry_center"),
     CONFIG_CENTER("config_center");
-    
+
```
Comment (kimmking) [inline sharding-orchestration/sharding-orchestration-core/sharding-orchestration-core-common/src/main/java/org/apache/shardingsphere/orchestration/core/common/CenterType.java]: blank line should be with four space

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-zookeeper-curator/src/main/java/org/apache/shardingsphere/orchestration/center/instance/CuratorZookeeperCenterRepository.java [hunk 7]
```diff
@@ -228,7 +228,7 @@ public void watch(final String key, final DataChangedEventListener dataChangedEv
             }
             DataChangedEvent.ChangedType changedType = getChangedType(event);
             if (DataChangedEvent.ChangedType.IGNORED != changedType) {
-                dataChangedEventListener.onChange(new DataChangedEvent(data.getPath(), null == data.getData() ? null : new String(data.getData(), "UTF-8"), changedType));
+                dataChangedEventListener.onChange(new DataChangedEvent(data.getPath(), null == data.getData() ? null : new String(data.getData(), Charsets.UTF_8), changedType));
```
Comment (kimmking) [inline sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-zookeeper-curator/src/main/java/org/apache/shardingsphere/orchestration/center/instance/CuratorZookeeperCenterRepository.java]: don't refer guava

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-zookeeper-curator/src/main/java/org/apache/shardingsphere/orchestration/center/instance/CuratorZookeeperCenterRepository.java [hunk 8]
```diff
@@ -86,7 +86,7 @@ private CuratorFramework buildCuratorClient(final CenterConfiguration config, fi
             builder.connectionTimeoutMs(operationTimeoutMilliseconds);
         }
         if (!Strings.isNullOrEmpty(digest)) {
-            builder.authorization("digest", digest.getBytes(Charsets.UTF_8))
+            builder.authorization("digest", digest.getBytes(StandardCharsets.UTF_8))
```
Comment (kimmking) [inline sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-zookeeper-curator/src/main/java/org/apache/shardingsphere/orchestration/center/instance/CuratorZookeeperCenterRepository.java]: remove nio static, use local static field.

File: sharding-orchestration/sharding-orchestration-core/sharding-orchestration-core-common/src/main/java/org/apache/shardingsphere/orchestration/core/common/CenterType.java [hunk 9]
```diff
@@ -31,16 +32,18 @@
...
     private final String value;
-    
+
```
Comment (kimmking) [inline sharding-orchestration/sharding-orchestration-core/sharding-orchestration-core-common/src/main/java/org/apache/shardingsphere/orchestration/core/common/CenterType.java]: blank line should be with four space

**Comment by tristaZero** [comment]
It is rather work-hard coding and reviewing from the thread of comments.
Reviewing carefully @kimmking and polishing with patience @yu199195 made this PR so great. 👍

**Pull request merged**